Add testing and fix `weights_only` load for quantized types and nn.Parameters with python attrs

mikaylagawarecki opened the pull request:
Adds the following to allowed globals for the `weights_only` unpickler
- [x] `torch._utils._rebuild_qtensor` and qtensor related types
- [x] `torch._utils._rebuild_parameter_with_state` (used deserializing a parameter that has user-defined attributes like `Param.foo`)

The remaining rebuild functions that have not been allowlisted are

- [x] `torch._utils._rebuild_wrapper_subclass` (allowlisted in above PR)
- [ ] `torch._utils._rebuild_device_tensor_from_numpy` 
- [ ] `torch._utils._rebuild_xla_tensor` (legacy)

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #124330

mikaylagawarecki committed:
Add testing and fix weights_only for quantized types and nn.Parameters with python attrs

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Add testing and fix weights_only for quantized types and nn.Parameters with python attrs"




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Add testing and fix `weights_only` load for quantized types and nn.Parameters with python attrs"

Adds the following to allowed globals for the `weights_only` unpickler
- [x] `torch._utils._rebuild_qtensor` and qtensor related types
- [x] `torch._utils._rebuild_parameter_with_state` (used deserializing a parameter that has user-defined attributes like `Param.foo`)

The remaining rebuild functions that have not been allowlisted are

- [x] `torch._utils._rebuild_wrapper_subclass` (allowlisted in above PR)
- [ ] `_rebuild_device_tensor_from_numpy` 
- [ ] `_rebuild_xla_tensor` (legacy)




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Add testing and fix `weights_only` load for quantized types and nn.Parameters with python attrs"

Adds the following to allowed globals for the `weights_only` unpickler
- [x] `torch._utils._rebuild_qtensor` and qtensor related types
- [x] `torch._utils._rebuild_parameter_with_state` (used deserializing a parameter that has user-defined attributes like `Param.foo`)

The remaining rebuild functions that have not been allowlisted are

- [x] `torch._utils._rebuild_wrapper_subclass` (allowlisted in above PR)
- [ ] `torch._utils._rebuild_device_tensor_from_numpy` 
- [ ] `torch._utils._rebuild_xla_tensor` (legacy)




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Add testing and fix `weights_only` load for quantized types and nn.Parameters with python attrs"

Adds the following to allowed globals for the `weights_only` unpickler
- [x] `torch._utils._rebuild_qtensor` and qtensor related types
- [x] `torch._utils._rebuild_parameter_with_state` (used deserializing a parameter that has user-defined attributes like `Param.foo`)

The remaining rebuild functions that have not been allowlisted are

- [x] `torch._utils._rebuild_wrapper_subclass` (allowlisted in above PR)
- [ ] `torch._utils._rebuild_device_tensor_from_numpy` 
- [ ] `torch._utils._rebuild_xla_tensor` (legacy)




[ghstack-poisoned]

albanD commented on torch/_weights_only_unpickler.py:
```diff
@@ -92,6 +92,12 @@ def _get_allowed_globals():
+        torch.qint32,
+        torch.quint8,
+        torch.quint4x2,
+        torch.quint2x4,
```
We should have that list somewhere shared. Maybe we should re-use `torch.storage._dtype_to_storage_type_map` ?

mikaylagawarecki replied:
This list only contains dtypes serialized with `rebuild_tensor_v2` added `torch.storage._new_types` for dtypes serialized with `_rebuild_tensor_v3`

albanD commented on torch/_weights_only_unpickler.py:
```diff
@@ -106,9 +112,20 @@ def _get_allowed_globals():
     for f in [
         torch._utils._rebuild_parameter,
+        torch._utils._rebuild_parameter_with_state,
+        torch._utils._rebuild_qtensor,
```
I guess you inspected all 7 functions here?

mikaylagawarecki committed:
Update on "Add testing and fix `weights_only` load for quantized types and nn.Parameters with python attrs"

Adds the following to allowed globals for the `weights_only` unpickler
- [x] `torch._utils._rebuild_qtensor` and qtensor related types
- [x] `torch._utils._rebuild_parameter_with_state` (used deserializing a parameter that has user-defined attributes like `Param.foo`)

The remaining rebuild functions that have not been allowlisted are

- [x] `torch._utils._rebuild_wrapper_subclass` (allowlisted in above PR)
- [ ] `torch._utils._rebuild_device_tensor_from_numpy` 
- [ ] `torch._utils._rebuild_xla_tensor` (legacy)




[ghstack-poisoned]

albanD reviewed (approved):
Thanks!

mikaylagawarecki commented:
@pytorchbot merge

mikaylagawarecki committed:
Update on "Add testing and fix `weights_only` load for quantized types and nn.Parameters with python attrs"

Adds the following to allowed globals for the `weights_only` unpickler
- [x] `torch._utils._rebuild_qtensor` and qtensor related types
- [x] `torch._utils._rebuild_parameter_with_state` (used deserializing a parameter that has user-defined attributes like `Param.foo`)

The remaining rebuild functions that have not been allowlisted are

- [x] `torch._utils._rebuild_wrapper_subclass` (allowlisted in above PR)
- [ ] `torch._utils._rebuild_device_tensor_from_numpy` 
- [ ] `torch._utils._rebuild_xla_tensor` (legacy)




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Add testing and fix `weights_only` load for quantized types and nn.Parameters with python attrs"

Adds the following to allowed globals for the `weights_only` unpickler
- [x] `torch._utils._rebuild_qtensor` and qtensor related types
- [x] `torch._utils._rebuild_parameter_with_state` (used deserializing a parameter that has user-defined attributes like `Param.foo`)

The remaining rebuild functions that have not been allowlisted are

- [x] `torch._utils._rebuild_wrapper_subclass` (allowlisted in above PR)
- [ ] `torch._utils._rebuild_device_tensor_from_numpy` 
- [ ] `torch._utils._rebuild_xla_tensor` (legacy)




[ghstack-poisoned]

mikaylagawarecki commented:
@pytorchbot merge
